app/src/main/java/com/trongdeptrai/soundcloud/screen/home/HomeFragment.java

```diff
@@ -37,9 +47,35 @@ public void initView(View v) {
 
     @Override
     public void initData() {
+        TrackLocalDataSource trackLocalDataSource = TrackLocalDataSource.getInstance();
+        TrackRemoteDataSource trackRemoteDataSource = TrackRemoteDataSource.getInstance();
+        TrackRepository trackRepo =
+                TrackRepository.getInstance(trackLocalDataSource, trackRemoteDataSource);
+        HomeContract.Presenter presenter = new HomePresenter(trackRepo);
+        presenter.setView(this);
+        presenter.getTrackByGenre(Genres.ALL_MUSIC);
+        presenter.getTrackByGenre(Genres.ALL_AUDIO);
+        presenter.getTrackByGenre(Genres.ALTERNATIVE_ROCK);
+        presenter.getTrackByGenre(Genres.AMBIENT);
+        presenter.getTrackByGenre(Genres.CLASSICAL);
+        presenter.getTrackByGenre(Genres.COUNTRY);
     }
 
     @Override
     public void initListener() {
     }
+
+    @Override
+    public void onGetTrendingTrackSucceed(List<Track> data) {
+    }
+
+    @Override
+    public void onGetTrackByGenresSucceed(List<Track> data, String genre) {
+        new ArrayList<Genre>().add(new Genre(genre, data));
+    }
+
+    @Override
+    public void onError(Exception e) {
+        Toast.makeText(getContext(), TOAST_ERROR, Toast.LENGTH_SHORT).show();
```

Comment:
nên show lỗi từ http trả về e
```suggestion
        Toast.makeText(getContext(), e.getMessage(), Toast.LENGTH_SHORT).show();
```